web/app.js

```diff
@@ -1550,6 +1550,11 @@ const PDFViewerApplication = {
       this.setTitle(this._contentDispositionFilename);
     }
 
+    if (pdfDocument.isPureXfa && AppOptions.get("annotationEditorEnabled")) {
+      document.getElementById("editorModeButtons").classList.add("hidden");
+      console.warn("XFA: Editing is not implemented.");
+    }
+
```

Comment:
I believe that we need to modify the `BaseViewer` instead, such that we actually disable editing properly:
```diff
--- a/web/base_viewer.js
+++ b/web/base_viewer.js
@@ -719,9 +719,8 @@ class BaseViewer {
         const annotationLayerFactory =
           this.#annotationMode !== AnnotationMode.DISABLE ? this : null;
         const xfaLayerFactory = isPureXfa ? this : null;
-        const annotationEditorLayerFactory = this.#annotationEditorUIManage
-          ? this
-          : null;
+        const annotationEditorLayerFactory =
+          this.#annotationEditorUIManager && !isPureXfa ? this : null;
```

(And we definitely need a better way to hide these buttons, since doing DOM-lookups like this isn't great; I'll try to look into that hopefully tomorrow.)